### PR TITLE
Add improved handling of pil

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Should solve the deprecation problem with PIL:
- https://github.com/huggingface/diffusers/pull/788
- https://github.com/huggingface/diffusers/issues/784
- https://github.com/huggingface/diffusers/pull/1259

Co-authored-by: @michellemoorre
Co-authored-by: @kashif
Co-authored-by: @allo-
Co-authored-by: @Thomas-MMJ

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -78,7 +78,7 @@
 # 1. all dependencies should be listed here with their version requirements if any
 # 2. once modified, run: `make deps_table_update` to update src/diffusers/dependency_versions_table.py
 _deps = [
-    "Pillow<10.0",  # keep the PIL.Image.Resampling deprecation away
+    "Pillow",  # keep the PIL.Image.Resampling deprecation away
```
Comment (pcuenca) [inline setup.py:81]:
```suggestion
    "Pillow",
```

File: src/diffusers/utils/pil_utils.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        "nearest": PIL.Image.Resampling.NEAREST,
+    }
+else:
+    PIL_INTERPOLATION = {
+        "linear": PIL.Image.LINEAR,
+        "bilinear": PIL.Image.BILINEAR,
+        "bicubic": PIL.Image.BICUBIC,
+        "lanczos": PIL.Image.LANCZOS,
+        "nearest": PIL.Image.NEAREST,
+    }
```
Comment (pcuenca) [inline src/diffusers/utils/pil_utils.py:21]: I'd prefer these to be symbols to prevent mistakes and allow autocompletion. Can we use `Enum`s  maybe?

File: docs/source/api/pipelines/latent_diffusion.mdx [hunk 3]
```diff
@@ -39,9 +39,9 @@ The original codebase can be found [here](https://github.com/CompVis/latent-diff
 
 
 ## LDMTextToImagePipeline
-[[autodoc]] pipelines.latent_diffusion.pipeline_latent_diffusion.LDMTextToImagePipeline
+[[autodoc]] LDMTextToImagePipeline
```
Comment (pcuenca) [inline docs/source/api/pipelines/latent_diffusion.mdx:42]: This is not related to this PR is it? Fine for me either way :)

**Review by pcuenca — APPROVED** [review]
There are a few changes about latent diffusion that are not related to PIL as far as I can tell, but I'm ok with that :)

File: src/diffusers/utils/pil_utils.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/diffusers/utils/pil_utils.py:21]: Makes sense!

Comment (patrickvonplaten) [reply src/diffusers/utils/pil_utils.py:21]: Actually, no this will lead to a weird API then because they are functions that are not used to compare anything. I don't want to add `if ...` comparisons everywhere in the code. Think it's actually fine as is. 

E.g. If I do:

```python
class PIL_INTERPOLATION(Enum): 
    LINEAR = PIL.Image.Resampling.BILINEAR
```

Then one **cannot** call the function with PIL_INTERPOLATION.LINEAR only with PIL_INTERPOLATION.LINEAR.value which is weird. Actually keen to leave as is here -> we have the same logic in `transformers` for the activation functions: https://github.com/huggingface/transformers/blob/4fb34de99e800689862f80cd618e6468fcb7b28b/src/transformers/activations.py#L152

Comment (patrickvonplaten) [reply src/diffusers/utils/pil_utils.py:21]: Wdyt @pcuenca ?

Comment (pcuenca) [reply src/diffusers/utils/pil_utils.py:21]: Oh, that's weird, I thought it would work without `.value`. That's too bad, let's keep them as strings then.

**Review by patil-suraj — APPROVED** [review]
Looks good!

**Comment by patil-suraj** [comment]
(Note, the `Co-authored-by:` should go in the PR commit description when merging the PR)

**Comment by patrickvonplaten** [comment]
> (Note, the `Co-authored-by:` should go in the PR commit description when merging the PR)

Can't find the email addresses now so just an in-official co-authoring :sweat_smile: But think that's ok given that none of the PRs were finished
